kibana.md

```diff
@@ -1101,3 +1101,189 @@ an admin, who has just configured access for a new user. They would like
+    });
+  }
+
+  return next();
+}
+```
+We will pass new logo custom metadata when logged in user username is 'admin'
+
+
+2. To replace the logo, we need to declare the custom kibana js file `readonlyrest_kbn.kibana_custom_js_inject_file: '/path/to/custom_kibana.js'`
+
+```js
+const logoHeader = document.querySelector('.euiHeaderLogo');
+
+if (window.ROR_METADATA.newLogo) {
+  Array.from(logoHeader.childNodes).forEach(node => {
+    node.style.display = 'none';
+  });
+
+  const observer = new MutationObserver(mutations => {
+    mutations.forEach(mutation => {
+      mutation.addedNodes.forEach(node => {
+        const customLogo = document.querySelector('#customLogo');
+
+        const createCustomLogo = () => {
+          const img = document.createElement('img');
+          img.src = `data:image/svg+xml;base64,${window.ROR_METADATA.newLogo}`;
+          img.style.width = '32px';
+          img.style.height = '32px';
+          img.id = 'customLogo';
+          logoHeader.appendChild(img);
+        };
+
+        const hideAllLogoElements = () => {
+          Array.from(logoHeader.childNodes).forEach(node => {
+            node.style.display = 'none';
+          });
+        };
+
+        const handleInit = () => {
+          hideAllLogoElements();
+          createCustomLogo();
+        };
+
+        if (customLogo) {
+          const displayCustomLogo = () => {
+            customLogo.style.display = 'block';
+          };
+          const hideCustomLogo = () => {
+            customLogo.style.display = 'none';
+          };
+          if (node.role === 'progressbar') {
+            hideCustomLogo();
+          }
+
+          if (node.role === 'img') {
+            const hideDefaultLogo = () => {
+              node.style.display = 'none';
+            };
+
+            hideDefaultLogo();
+            displayCustomLogo();
+          }
+        }
+
+        if (node.dataset.type === 'logoElastic' && !customLogo) {
+          handleInit();
+        }
+      });
+    });
+  });
+
+  observer.observe(logoHeader, { childList: true });
+}
+```
+All metadata should be available via `window.ROR_METADATA`. You can use `window.ROR_METADATA.newLogo`. After login as a user with username `admin` you should see a custom logo
+
+#### Reject kibana machine2machine based on a custom metadata
```

Comment:
Reject machine to machine traffic using custom metadata ACL rules

Reply:
Done ✔️